### PR TITLE
UIOR-137 Error Modal: Order has inactive Vendor/Access Provider

### PR DESCRIPTION
**Pull request by AndreiKandratsenka**
## Purpose
it needs to provide error message in modal window
https://issues.folio.org/browse/UIOR-137

## Approach
extended functionality open PO response error modal,
extended update PO response error object 

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/PurchaseOrder/UpdateOrderErrorModal/UpdateOrderErrorModal.js [hunk 1]
```diff
@@ -25,15 +25,23 @@ const UpdateOrderErrorModal = ({ orderNumber, cancel, errorCode }) => {
...
+      ))}
     </Modal>
   );
 };
 
 UpdateOrderErrorModal.propTypes = {
   orderNumber: PropTypes.string.isRequired,
   cancel: PropTypes.func.isRequired,
-  errorCode: PropTypes.string.isRequired,
+  errors: PropTypes.arrayOf(PropTypes.object).isRequired,
```
Comment (aliaksei-chumakou) [inline src/components/PurchaseOrder/UpdateOrderErrorModal/UpdateOrderErrorModal.js]: Some misleading logic here, prop is required, but you set a default value above `errors = []`

File: test/bigtest/tests/PurchaseOrder/update-order-error-modal-test.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ describe('Update Order Error modal', () => {
         <UpdateOrderErrorModal
           cancel={cancelFake}
           orderNumber="123"
-          errorCode="vendorIsInactive"
+          errors={[{ code: 'vendorIsInactive' }]}
```
Comment (aliaksei-chumakou) [inline test/bigtest/tests/PurchaseOrder/update-order-error-modal-test.js]: It looks like a good place to use constant

File: src/components/Utils/order/showUpdateOrderError.js [hunk 3]
```diff
@@ -21,15 +21,31 @@ const showUpdateOrderError = async (response, callout, openModal) => {
...
+    case ERROR_CODES.vendorIsInactive:
+    case ERROR_CODES.vendorNotFound: {
+      openModal([{ code }]);
+      break;
+    }
+    case ERROR_CODES.accessProviderIsInactive:
+    case ERROR_CODES.accessProviderNotFound: {
+      let errors =
+        get(error, 'errors.0.parameters', [])
+          .filter(({ key }) => key === 'poLineNumber')
```
Comment (aliaksei-chumakou) [inline src/components/Utils/order/showUpdateOrderError.js]: magic string `poLineNumber`

**Comment by aliaksei-chumakou** [comment]
mostly looks good, IMHO. Just some clean-up would be good

File: src/components/PurchaseOrder/UpdateOrderErrorModal/UpdateOrderErrorModal.js (hunk 1 above)
Comment (AndreiKandratsenka) [reply src/components/PurchaseOrder/UpdateOrderErrorModal/UpdateOrderErrorModal.js]: updated

File: test/bigtest/tests/PurchaseOrder/update-order-error-modal-test.js (hunk 2 above)
Comment (AndreiKandratsenka) [reply test/bigtest/tests/PurchaseOrder/update-order-error-modal-test.js]: fixed

File: src/components/Utils/order/showUpdateOrderError.js (hunk 3 above)
Comment (AndreiKandratsenka) [reply src/components/Utils/order/showUpdateOrderError.js]: fixed

File: test/bigtest/tests/Utils/order/show-update-order-error-test.js [hunk 4]
```diff
@@ -3,18 +3,24 @@ import {
...
-    .keys(ERROR_CODES)
-    .filter(item => item !== ERROR_CODES.vendorIsInactive && item !== ERROR_CODES.vendorNotFound);
+const CALLOUT_ERROR_CODES = omit(
+  ERROR_CODES,
+  [
+    'vendorIsInactive',
+    'accessProviderIsInactive',
+    'vendorNotFound',
+    'accessProviderNotFound',
+  ] // eslint-disable-line comma-dangle
```
Comment (NikitaSedyx) [inline test/bigtest/tests/Utils/order/show-update-order-error-test.js]: don't see reason why we need to disable eslint here, `,` is ok for me

**Pull request merged**